expyriment/design/_structure.py

```diff
@@ -2072,9 +2072,7 @@ def find_stimulus(self, id):
     def copy(self):
         """Return a copy of the trial."""
 
-        stimlist = []
-        for stim in self._stimuli:
-            stimlist.append(stim)
```

Comment:
I fail to understand why you make a shallow copy of `self._stimuli`.

I understand this code resets `self._stimuli` before performing the deep copy of `self`, to avoid copying the stimuli. But why the prior shallow copy of `self._stimuli`?

I think the following code does what you want without requiring a shallow copy before the deep copy:
```suggestion
        stimlist = self._stimuli
```

Reply:
```
In [20]: stims = [xpy.stimuli.TextLine("Hello"), xpy.stimuli.TextLine("World")]
Warning: Stimulus created before initializing (experiment defaults won't apply)!
Warning: Stimulus created before initializing (experiment defaults won't apply)!

In [21]: stimlist = []

In [22]: for stim in stims:
    ...:     stimlist.append(stim)
    ...: 

In [23]: stimlist2 = stims

In [24]: stims.append(xpy.stimuli.TextLine("Another one"))
Warning: Stimulus created before initializing (experiment defaults won't apply)!

In [25]: stimlist
Out[25]: 
[<expyriment.stimuli._textline.TextLine at 0x10a921a70>,
 <expyriment.stimuli._textline.TextLine at 0x10a922190>]

In [26]: stimlist2
Out[26]: 
[<expyriment.stimuli._textline.TextLine at 0x10a921a70>,
 <expyriment.stimuli._textline.TextLine at 0x10a922190>,
 <expyriment.stimuli._textline.TextLine at 0x107b62b10>]
```

Reply:
I fail to understand the message you're trying to convey. I know perfectly well the difference between:
* assigning a reference to a list,
* performing a shallow copy of a list (`copy()`),
* performing a deep copy of a list (`deepcopy()`)

It's a bit cavalier to close the PR like this. I'll open a new one to try to better explain my point, because it looks like you didn't understand it.

Reply:
I suggest using `stimlist = self._stimuli` instead of `stimlist = self._stimuli.copy()` because **you don't need a copy** (shallow or deep) of `self._stimuli` at this point. You just need to keep a reference to the initial list.